src/coreclr/interpreter/compiler.cpp

```diff
@@ -2508,15 +2508,15 @@ void InterpCompiler::EmitCall(CORINFO_RESOLVED_TOKEN* constrainedClass, bool rea
             }
             else if ((callInfo.classFlags & CORINFO_FLG_ARRAY) && !readonly)
```

Comment:
```suggestion
            else if (callInfo.classFlags & CORINFO_FLG_ARRAY)
```
`!readonly` should not be needed here

Reply:
Instead, you can check for newobj opcode and then you do not need the condition below:

`if ((callInfo.classFlags & CORINFO_FLG_ARRAY) && newObj)`